src/Components/Aspire.Confluent.Kafka/README.md

```diff
@@ -0,0 +1,157 @@
+
+### Use configuration providers
+
+The .NET Aspire Confluent Kafka component supports [Microsoft.Extensions.Configuration](https://learn.microsoft.com/dotnet/api/microsoft.extensions.configuration). It loads the `KafkaProducerSettings` or `KafkaConsumerSettings` from configuration by respectively using the `Aspire:Confluent:Kafka:Producer` and `Aspire.Confluent:Kafka:Consumer` keys. Example `appsettings.json` that configures some of the options:
+
+```json
+{
+  "Aspire": {
+    "Confluent": {
+      "Kafka": {
+        "Producer": {
+          "HealthChecks": true,
+          "Config": {
+            "Acks": "All"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The `Config` properties of both  `Aspire:Confluent:Kafka:Producer` and `Aspire.Confluent:Kafka:Consumer` configuration sections respectively bind to instances of [`ProducerConfig`](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ProducerConfig.html) and [`ConsumerConfig`](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ConsumerConfig.html).
+
+`Confluent.Kafka.Consumer<TKey, TValue>` requires the `ClientId` property to be set to let the broker track consumed message offsets.
+
+### Use inline delegates to configure `KafkaProducerSettings` and `KafkaConsumerSettings`.
+
+Also you can pass the `Action<KafkaProducerSettings> configureSettings` delegate to set up some or all the options inline, for example to disable health checks from code:
+
+```csharp
+    builder.AddKafkaProducer<string, string>("messaging", settings => settings.HealthChecks = false);
+```
+
+Similarly you can configure inline a consumer from code:
+```c#
+    builder.AddKafkaConsumer<string, string>("messaging", settings => settings.HealthChecks = false);
+```
+
+### Use inline delegates to configure `ProducerBuilder<TKey, TValue>` and `ConsumerBuilder<TKey, TValue>`.
+
+To configure `Confluent.Kafka` builders (for example to setup custom serializers/deserializers for message key and value) you can pass an `Action<ProducerBuilder<TKey, TValue>>` (or `Action<ConsumerBuilder<TKey, TValue>>`) from code:
+```c#
+    builder.AddKafkaProducer<string, MyMessage>("messaging", producerBuilder => {
+      producerBuilder.SetValueSerializer(new MyMessageSerializer());
+    })
+```
+
+You can refer to [`ProducerBuilder<TKey, TValue>`](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ProducerBuilder-2.html) and [`ConsumerBuilder<TKey, TValue>`](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.ConsumerBuilder-2.html) api documentation for more information.
+
+## AppHost extensions
+
+In your AppHost project, register an Apache Kafka container and consume the connection using the following methods:
+
+```csharp
+var messaging = builder.AddKafkaContainer("messaging");
```

Comment:
```suggestion
var messaging = builder.AddKafka("messaging");
```

In general, we recommend the non-container resource.

Reply:
done